validator/client/src/main/java/tech/pegasys/teku/validator/client/ValidatorStatusLogger.java

```diff
@@ -0,0 +1,135 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.api.response.v1.beacon.ValidatorStatus;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.validator.api.ValidatorApiChannel;
+
+public class ValidatorStatusLogger {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final int VALIDATOR_KEYS_PRINT_LIMIT = 20;
+
+  final Set<BLSPublicKey> validatorPublicKeys;
+  final ValidatorApiChannel validatorApiChannel;
+  final AtomicReference<Map<BLSPublicKey, ValidatorStatus>> latestValidatorStatuses =
+      new AtomicReference<>();
+
+  public ValidatorStatusLogger(
+      Set<BLSPublicKey> validatorPublicKeys, ValidatorApiChannel validatorApiChannel) {
+    this.validatorPublicKeys = validatorPublicKeys;
+    this.validatorApiChannel = validatorApiChannel;
+  }
+
+  public void printInitialValidatorStatuses() {
+    validatorApiChannel
+        .getValidatorStatuses(getAsIdentifiers(validatorPublicKeys))
+        .thenAccept(
+            maybeValidatorStatuses -> {
+              if (maybeValidatorStatuses.isEmpty()) {
+                LOG.error("Unable to retrieve validator statuses from BeaconNode.");
+                return;
+              }
+
+              Map<BLSPublicKey, ValidatorStatus> validatorStatuses = maybeValidatorStatuses.get();
+              if (validatorPublicKeys.size() < VALIDATOR_KEYS_PRINT_LIMIT) {
+                printValidatorStatusesOneByOne(validatorStatuses);
+              } else {
+                printValidatorStatusSummary(validatorStatuses);
+              }
+            })
+        .reportExceptions();
+  }
+
+  private void printValidatorStatusesOneByOne(
+      Map<BLSPublicKey, ValidatorStatus> validatorStatuses) {
+    for (BLSPublicKey publicKey : validatorPublicKeys) {
+      Optional<ValidatorStatus> maybeValidatorStatus =
+          Optional.ofNullable(validatorStatuses.get(publicKey));
+      maybeValidatorStatus.ifPresentOrElse(
+          validatorStatus ->
+              LOG.info(
+                  "Validator {} status is " + validatorStatus, publicKey.toAbbreviatedString()),
+          () -> LOG.error("Error retrieving status for validator {}", publicKey));
```

Comment:
Maybe just a warning since these are printing regularly and this won't interfere with functioning of the node?
```suggestion
          () -> LOG.warning("Unable to retrieve status for validator {}", publicKey));
```

Reply:
Done.